builder/builder.go

```diff
@@ -1640,6 +1647,64 @@ func (b *Builder) BuildImage(format string, template string) error {
 	return imagecmd.Run()
 }
 
+type dnfRepoConf struct {
+	RepoName, RepoURL string
+}
+
+const dnfConfRepoTemplate = `
+[{{.RepoName}}]
+name={{.RepoName}}
+failovermethod=priority
+baseurl={{.RepoURL}}
+enabled=1
+gpgcheck=0
+priority=1
+`
+
+// AddRepo adds and enables a repo configuration named repoInfo[0] pointing at
+// URL repoInfo[1]. It calls b.newConfIfNeeded() to create the DNF config if it
+// does not exist and performs a check to see if the repo passed has already
+// been configured.
+func (b *Builder) AddRepo(repoInfo []string) error {
+	repo := dnfRepoConf{
+		RepoName: repoInfo[0],
+		RepoURL:  repoInfo[1],
+	}
+
+	if err := b.newConfIfNeeded(); err != nil {
+		return err
+	}
+
+	f, err := os.OpenFile(b.Config.Builder.DNFConf, os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		helpers.PrintError(err)
+		return err
+	}
+
+	read, err := ioutil.ReadFile(b.Config.Builder.DNFConf)
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(string(read), fmt.Sprintf("[%s]", repo.RepoName)) {
```

Comment:
Since the dnf conf is an INI file, I would prefer to parse the INI and see if the appropriate section exists.

Reply:
Because `[arbitrary-string]` could appear within a quoted string, and by parsing the file as INI, you would not have to worry about this possibility.

Reply:
T_T fine